lib/elixir/lib/regex.ex

```diff
@@ -397,30 +404,40 @@ defmodule Regex do
   defp parts_to_index(:infinity),                      do: 0
   defp parts_to_index(n) when is_integer(n) and n > 0, do: n
 
-  defp do_split(_, string, offset, _counter, true) when byte_size(string) <= offset,
+  defp do_split(_, string, offset, _counter, true, _with_captures) when byte_size(string) <= offset,
     do: []
 
-  defp do_split(_, string, offset, 1, _trim),
+  defp do_split(_, string, offset, 1, _trim, _with_captures),
     do: [binary_part(string, offset, byte_size(string) - offset)]
 
-  defp do_split([], string, offset, _counter, _trim),
+  defp do_split([], string, offset, _counter, _trim, _with_captures),
     do: [binary_part(string, offset, byte_size(string) - offset)]
 
-  defp do_split([[{pos, _}|h]|t], string, offset, counter, trim) when pos - offset < 0,
-    do: do_split([h|t], string, offset, counter, trim)
+  defp do_split([[{pos, _}|h]|t], string, offset, counter, trim, with_captures) when pos - offset < 0,
+    do: do_split([h|t], string, offset, counter, trim, with_captures)
 
-  defp do_split([[]|t], string, offset, counter, trim),
-    do: do_split(t, string, offset, counter, trim)
+  defp do_split([[]|t], string, offset, counter, trim, with_captures),
+    do: do_split(t, string, offset, counter, trim, with_captures)
 
-  defp do_split([[{pos, length}|h]|t], string, offset, counter, trim) do
+  defp do_split([[{pos, length}|h]|t], string, offset, counter, trim, with_captures) do
```

Comment:
I thought about that, but the differences between the two functions seems too small to be that redundant. Another choice could be matching the truth table of `{trim, with_captures}`. Should I?